pkg/storage/s3/s3.go

```diff
@@ -437,12 +437,22 @@ func (d *driver) RemoveStorage(cr *opapi.Config, modified *bool) error {
 		return err
 	}
 
+	// Wait until the bucket does not exist
+	if err := svc.WaitUntilBucketNotExists(&s3.HeadBucketInput{
+		Bucket: aws.String(d.Config.Bucket),
+	}); err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			util.UpdateCondition(cr, opapi.StorageExists, operatorapi.ConditionTrue, aerr.Code(), aerr.Error(), modified)
+		}
+
+		return err
+	}
+
 	cr.Spec.Storage.S3.Bucket = ""
 	d.Config.Bucket = ""
 	cr.Status.Storage.S3 = d.Config.DeepCopy()
-	if d.StorageChanged(cr, modified) {
```

Comment:
Pretty sure that this was keeping the cr from being updated since we removed the bucket from the Spec and the Status